src/Foundation/NSMutableArray_1.cs

```diff
@@ -166,7 +166,7 @@ void ValidateIndex (nint index)
 			if (index < 0)
 				throw new IndexOutOfRangeException (nameof (index));
 
-			if ((nuint) index >= Count)
+			if ((nuint) index > Count)
```

Comment:
This check makes the < 0 check redundant btw.

Reply:
It's used in a lot of places in corefx: [1](https://source.dot.net/#System.Collections/System/Collections/BitArray.cs,0ed1a118969f6946,references) [2](https://source.dot.net/#System.Memory/System/Buffers/ArrayMemoryPool.cs,fabbb5d9ac585197,references) [3](https://source.dot.net/#System.Memory/System/Buffers/SequenceReader.cs,327) [4](https://source.dot.net/#System.Memory/System/Buffers/Text/Base64Encoder.cs,722f85d379c418d5,references) [5 List<T>](https://source.dot.net/#System.Private.CoreLib/shared/System/Collections/Generic/List.cs,0c65bec4d3fac21e,references) [6 List<T>.InsertRange](https://source.dot.net/#System.Private.CoreLib/shared/System/Collections/Generic/List.cs,79de3e39e69a4811,references)